nengo_loihi/loihi_cx.py

```diff
@@ -173,19 +173,24 @@ def configure_filter(self, tau_s, dt=0.001, default=False):
             Simulator time step.
         default : bool
             Whether we are setting the default. Will only override the
-            current value if not explicitly set (i.e. if ``not decayU_set``).
+            current value if not explicitly set (i.e. if ``tau_s is None``).
         """
-        if default and self.decayU_set:
+        if default and self.tau_s is not None:
             return
 
-        decayU = 1 if tau_s == 0 else -np.expm1(-dt/np.asarray(tau_s))
-        if self.decayU_set and not np.allclose(decayU, self.decayU):
-            raise BuildError(
-                "Cannot change tau_s on already configured neurons")
+        if self.tau_s is not None and tau_s < self.tau_s:
+            warnings.warn("tau_s is already set to %g, which is larger than "
+                          "%g. Using %g." % (self.tau_s, tau_s, self.tau_s))
+            return
+        elif self.tau_s is not None and tau_s > self.tau_s:
+            warnings.warn(
+                "tau_s is currently %g, which is smaller than %g. Overwriting "
+                "tau_s with %g." % (self.tau_s, tau_s, tau_s))
 
+        decayU = 1 if tau_s == 0 else -np.expm1(-dt/np.asarray(tau_s))
         self.decayU[:] = decayU
         self.scaleU = decayU > self.DECAY_SCALE_TH
-        self.decayU_set = not default
```

Comment:
So the idea behind the `default` value is that it initializes `decayU`, but in a way where we won't get mad if that value is overridden later. I'm worried that now, we're going to throw more warnings, because we're going to warn even if the user overwrites the default. I don't think we should do this, but I agree, it is annoying to track whether it has been set. Perhaps if `default=True`, we don't set `tau_s`, we only set `decayU`. Then, if something overwrites that default, we won't complain. Does that make sense?

Reply:
Just reading your comment above. This would solve those errors, too.

Reply:
That third paragraph in the comment above is essentially a response to that question... I was going to only set `self.tau_s` if we're not setting the default, but then we can't tell if the default has been set yet (because right now we check `if self.tau_s is not None`). Is there another way to tell if the default's been set?

Reply:
Why do we need to know if the default has been set? We didn't need to know this before. The point is that it's just a default, and we don't care if it gets overridden.

Reply:
Previously we weren't allowing non-defaults to be overridden, it would just raise an exception, so there was only one case: overriding a default. Now, there are two cases, overriding a default, and overriding a value explicitly set, and we need some way of differentiating between those two.